UnitTests/PayPalTests/PayPalClient_Tests.swift

```diff
@@ -0,0 +1,26 @@
+import XCTest
+import PayPalCheckout
+@testable import PaymentsCore
+@testable import PayPal
+
+class PayPalClient_Tests: XCTestCase {
+
+// TODO: Put up a separate PR with Mocks for NXO so we can test the below things
```

Comment:
Jax & I spent a solid amount of time trying to mock the `CheckoutConfig`, which we were eventually able to sort of get working via a protocol. Mocking the `Checkout.set` is hard to test since this is a static method on a class, and not on an instance of a class. 

So .... there's some work to be done here. Either on the NXO side making an API that is more easily testable, or spending more time trying to come up with a way to mock this out, it will be wonky though. I am not sure how we'd do it.

I think it warrants a larger convo around NXO testing strategy & setup, and how that affects the Northstar SDK.

Reply:
Yeah NXO is not built in a way that makes it super easy to test. I spent some time hacking on this - i think this might work for us for now, to let us test / mock out the functionality of the `Checkout` type:

```swift
protocol PayPalUIFlow {

  static func set(config: CheckoutConfig)

  static func start(
    presentingViewController: UIViewController?,
    createOrder: CheckoutConfig.CreateOrderCallback?,
    onApprove: CheckoutConfig.ApprovalCallback?,
    onShippingChange: CheckoutConfig.ShippingChangeCallback?,
    onCancel: CheckoutConfig.CancelCallback?,
    onError: CheckoutConfig.ErrorCallback?
  )
}

extension Checkout: PayPalUIFlow { }

class PayPalClient2 {
  let CheckoutFlow: PayPalUIFlow.Type

  init(flow: PayPalUIFlow.Type) {
    self.CheckoutFlow = flow
  }

  func configCheckout() {
    CheckoutFlow.set(config: CheckoutConfig(clientID: "", returnUrl: ""))
  }

  func startCheckout() {
    CheckoutFlow.start(
      presentingViewController: nil,
      createOrder: nil,
      onApprove: nil,
      onShippingChange: nil,
      onCancel: nil,
      onError: nil
    )
  }
}

class CheckoutMock: PayPalUIFlow {
  static func set(config: CheckoutConfig) {
    // no-op
  }

  static func start(presentingViewController: UIViewController?, createOrder: CheckoutConfig.CreateOrderCallback?, onApprove: CheckoutConfig.ApprovalCallback?, onShippingChange: CheckoutConfig.ShippingChangeCallback?, onCancel: CheckoutConfig.CancelCallback?, onError: CheckoutConfig.ErrorCallback?) {
    // no-op
  }
}


let payPalClient = PayPalClient2(flow: Checkout.self)

let payPalMockClient = PayPalClient2(flow: CheckoutMock.self)

```

Reply:
Awesome, yeh this makes sense!

Reply:
We also might have to mock `CheckoutConfig` but what you have above is most of the work.